app/builders/immediate_email_builder.rb

```diff
@@ -15,47 +15,56 @@ def call
 
 private
 
-  attr_reader :subscriber_content_changes
+  attr_reader :subscription_content_changes
 
   def records
-    subscriber_content_changes.map do |subscriber_content_change|
-      single_email_record(subscriber_content_change)
+    subscription_content_changes.map do |subscription_content_change|
+      single_email_record(subscription_content_change)
     end
   end
 
-  def single_email_record(subscriber_content_change)
-    subscriber = subscriber_content_change.fetch(:subscriber)
-    content_change = subscriber_content_change.fetch(:content_change)
+  def single_email_record(subscription_content_change)
+    content_change = subscription_content_change.fetch(:content_change)
+    subscription = subscription_content_change[:subscription]
+
+    subscriber = if subscription
```

Comment:
When might we not have a `subscription` in the `subscription_content_change`?

Also, I wonder if/whether it's a good idea to take advantage of passing a default to `.fetch` rather than the `if` statement? Maybe it would look a bit unwieldy given the length of `subscription_content_change`.

Reply:
If we're sending courtesy emails, we don't have a subscription available, only a subscriber (the courtesy email address subscriber).

Reply:
The reason I chose to not use `fetch` with a default argument is that it means a user could override the `subscriber` if they pass a hash that includes both a `subscription` and a `subscriber` which doesn't feel like something that should be possible to do.

I suppose we could catch that and raise an exception?

Reply:
that sounds sensible enough. I would say leave it as it is then 👍